pkg/apis/v1/dnsrecord_types.go

```diff
@@ -123,10 +130,45 @@ type DNSRecordStatus struct {
 	// needs to retry the update for that specific zone.
 	// +optional
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// endpoints are the last endpoints that were successfully published by the provider
+	//
+	// Provides a simple mechanism to store the current provider records in order to
+	// delete any that are no longer present in DNSRecordSpec.Endpoints
+	//
+	// Note: This will not be required if/when we switch to using external-dns since when
+	// running with a "sync" policy it will clean up unused records automatically.
+	Endpoints []*Endpoint `json:"endpoints,omitempty"`
```

Comment:
did we want to also store the nameservers for this zone in the status? So if a user wants to delegate a zone to us we can provide the needed nameservers for them to update their own DNS zone with NS records

Reply:
oh nvm this is the DNSRecord